### PR TITLE
Implement block programming (using Blockly)

### PR DESCRIPTION
**Pull request by okalachev**
User documentation: https://github.com/CopterExpress/clover/blob/blockly/docs/ru/blocks.md.

Internal documentation: https://github.com/CopterExpress/clover/blob/blockly/clover_blocks/README.md.

### CODE REVIEW

File: clover_blocks/package.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  <!-- Use buildtool_depend for build tool packages: -->
+  <!--   <buildtool_depend>catkin</buildtool_depend> -->
+  <!-- Use exec_depend for packages you need at runtime: -->
+  <!--   <exec_depend>message_runtime</exec_depend> -->
+  <!-- Use test_depend for packages you need only for testing: -->
+  <!--   <test_depend>gtest</test_depend> -->
+  <!-- Use doc_depend for packages you need only for building documentation: -->
+  <!--   <doc_depend>doxygen</doc_depend> -->
+  <buildtool_depend>catkin</buildtool_depend>
+
```
Comment (sfalexrog) [inline clover_blocks/package.xml:38]: You probably want to add additional dependencies here (`message_generation` as buildtime dependency, `message_runtime` as runtime dependency, `rospy` as a general dependency)

File: clover_blocks/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+generate_messages(
+  DEPENDENCIES
+  # std_msgs  # Or other packages containing msgs
+)
+
+catkin_package(
+#  INCLUDE_DIRS include
+#  LIBRARIES roslaunch_editor
+#  CATKIN_DEPENDS other_catkin_pkg
```
Comment (sfalexrog) [inline clover_blocks/CMakeLists.txt:24]: `CATKIN_DEPENDS` should not be commented and should have `message_runtime` as a dependency.

File: clover_blocks/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+#   PATTERN ".svn" EXCLUDE
+# )
+
+## Mark other files for installation (e.g. launch and bag files, etc.)
+# install(FILES
+#   # myfile1
+#   # myfile2
+#   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+# )
+
```
Comment (sfalexrog) [inline clover_blocks/CMakeLists.txt:69]: You should use `catkin_install_python` for node installation, see http://docs.ros.org/api/catkin/html/howto/format2/installing_python.html (this should seamlessly change the interpreter to `python3` in Noetic)

File: clover_blocks/www/blocks.js [hunk 4]
```diff
@@ -0,0 +1,481 @@
+const COLOR_FLIGHT = 293;
+const COLOR_STATE = 36;
+const COLOR_LED = 143;
+const COLOR_GPIO = 200;
+const DOCS_URL = 'https://clover.coex.tech/en/blocks.html';
```
Comment (sfalexrog) [inline clover_blocks/www/blocks.js:5]: This probably won't be accessible when we are connected to the drone; maybe we should reference a local copy?

**Review by sfalexrog — COMMENTED** [review]
I believe I can mostly comment on the "plumbing" side of things. Looks good on the whole!

File: clover_blocks/package.xml (hunk 1 above)
Comment (okalachev) [reply clover_blocks/package.xml:38]: Simply added `depend`s.

File: clover_blocks/www/blocks.js (hunk 4 above)
Comment (okalachev) [reply clover_blocks/www/blocks.js:5]: Reasonable. But can we rely this package is used on Clover's image/official VM but not autonomously?

`selfcheck.py` links to the external site.

File: clover_blocks/CMakeLists.txt (hunk 3 above)
Comment (okalachev) [reply clover_blocks/CMakeLists.txt:71]: Are you sure scripts won't run without this directive?

**Pull request merged**